### PR TITLE
Refactor `setup.rb` and update specs

### PR DESCRIPTION
**Pull request by kuldeepaggarwal**
@kristianmandrup: Please review.

### CODE REVIEW

File: lib/feature-config/feature/filter.rb [hunk 1]
```diff
@@ -3,6 +3,11 @@ def self.build_filters(attributes)
     attributes.map { |name, options| class_for(name).new(options) }
   end
 
+  def self.class_for(filter_name)
+    "#{name}/#{filter_name}".classify.constantize
+  end
+  private_class_method :class_for
```
Comment (kuldeepaggarwal) [inline lib/feature-config/feature/filter.rb:9]: I can also use the new syntax of ruby 2.2.x .

``` rb
private_class_method  def self.class_for(filter_name)
  # code
end
```

but I like the current implemented way.

**Comment by kristianmandrup** [comment]
You have really good coding style :) Did you read _Refactoring Ruby edition_?

**Comment by kuldeepaggarwal** [comment]
No, I haven't read any book. Is _Refactoring Ruby_ really a good book? Does it really help? Should I read it?

**Comment by kristianmandrup** [comment]
Looks like you already read most of it ;) The book I learned the most from regarding beautiful code and refactoring. Can highly recommend it. I can send you a free pdf version if you like ;)
Did you do any Node or ES6 programming?

**Pull request merged**